discordia/extensions.lua

```diff
@@ -137,9 +137,20 @@ end
 
 function string.split(str, delim)
 	if delim and delim ~= '' then
+		if find(str, delim) == nil then
+			return { str }
+		end
 		local words = {}
-		for word in gmatch(str .. delim, '(.-)' .. delim) do
+		local pattern = '(.-)' .. delim
+		local lastPos = 1
+		while true do
+			local _, pos, word = find(str, pattern, lastPos)
+			if not word then break end
 			insert(words, word)
+			lastPos = pos + 1
+		end
+		if lastPos <= #str then
```

Comment:
This is not necessary if you append `delim` to `str` before the loop, as in the original function.